src/MetadataUtility/Audio/Wave.cs

```diff
@@ -2,12 +2,156 @@
 // All code in this file and all associated files are the copyright and property of the QUT Ecoacoustics Research Group.
 // </copyright>
 
-namespace MetadataUtility.Audio.Vendors
+namespace MetadataUtility.Audio
 {
+    using LanguageExt;
+    using LanguageExt.Common;
+    using MetadataUtility.Utilities;
+
     public static class Wave
     {
+        public const string Mime = "audio/wave";
+
+        public const int WaveFileOffset = 7;
+        public const int WaveSampleRateOffset = 24;
+        public const int WaveChannelOffset = 22;
+        public const int WaveBitsPerSampleOffset = 34;
+        public const int WaveFileLengthOffset = 4;
+
         public static readonly byte[] WaveMagicNumber = new byte[] { (byte)'W', (byte)'A', (byte)'V', (byte)'E' };
 
+        public static readonly Error FileTooShort = Error.New("Error reading file: file is not long enough to have a duration header");
+
+        public static readonly Error FileTooShortWave = Error.New("Error reading file: file is not long enough to have a WAVE header");
+
         public static readonly byte[] DataBlockId = new byte[] { (byte)'d', (byte)'a', (byte)'t', (byte)'a' };
+
+        public static Fin<ulong> ReadTotalSamples(FileStream stream)
```

Comment:
All the methods in this class probably need to fail unless we're dealing with WAVE_FORMAT_PCM (or we can add support for other formats, but that is a huge can of worms)